packages/store/addon/-private/system/model/shim-model-class.ts

```diff
@@ -0,0 +1,63 @@
+import { map } from '@ember/object/computed';
+import CoreStore from '../core-store';
+import { stringify } from 'querystring';
+import { RelationshipSchema, AttributeSchema } from '../../ts-interfaces/record-data-schemas';
+import { Dict } from '../../ts-interfaces/utils';
+
+// Mimics the static apis of DSModel
+export default class ShimModelClass {
+  store: CoreStore;
+  modelName: string;
+  _id?: string;
+
+  // TODO Maybe expose the class here?
+  constructor(store: CoreStore, modelName: string, id?: string) {
+    this.store = store;
+    this.modelName = modelName;
+    this._id = id;
+  }
+
+  get fields(): Map<string, 'attribute' | 'belongsTo' | 'hasMany'> {
+    let attrs = this.store._attributesDefinitionFor(this.modelName, this._id);
+    let relationships = this.store._relationshipsDefinitionFor(this.modelName, this._id);
+    let fields = new Map<string, 'attribute' | 'belongsTo' | 'hasMany'>();
+    Object.keys(attrs).forEach(key => fields.set(key, 'attribute'));
+    Object.keys(relationships).forEach(key => fields.set(key, relationships[key]!.kind));
+    return fields;
+  }
+
+  get attributes(): Map<string, AttributeSchema> {
+    let attrs = this.store._attributesDefinitionFor(this.modelName, this._id);
+    //TODO Talk to mnorth about needing a cast
+    return new Map(Object.entries(attrs) as [string, AttributeSchema][]);
+  }
+
+  get relationshipsByName(): Map<string, RelationshipSchema> {
+    let relationships = this.store._relationshipsDefinitionFor(this.modelName, this._id);
+    //TODO Talk to mnorth about needing a cast
+    return new Map(Object.entries(relationships) as [string, RelationshipSchema][]);
```

Comment:
are we worried about these being uncached getters?

Reply:
if we can't cache we should document why we can't assume cacheable and how users should approach using these APIs when they are cacheable.